meetings/2021-06-30.md

```diff
@@ -0,0 +1,125 @@
+We've done that. Might be improved. But there is a pattern to start with.
+
+Matthieu: So it's not only subset of resource, it's also multiple resources.
+It's quite a departure from current authz, isn't it?
+
+Elf: There are discussions on shape masks.
+It's on the resource level to group them but also with virtual hierarchies based on relationships. We also use constructs to partition the relationships.
+But it's still assuming that we're linked to a resource.
+
+Henry: It's not that far from WAC since it has access to class for classes of resources that you can describe. Say read access to the class of all resources that are tagged over 18 or tagged as written by a group of people.
+
+Elf: https://github.com/solid/web-access-control-spec#aclaccesstoclass
+Just to clarify, that would be WAC+ because WAC doesn't support it.
+
+Henry: Of course, but it's there in the ontology so it could be used and because it's uncommonly used, it's harder to implement.
+The accessToClass just needs to be used. That's what you're describing.
+
+Elf: Just saying it's not supported by design in https://github.com/solid/web-access-control-spec
+
+Henry: The spec at the moment is extremely limited but it's not like we don't have to improve it.
+Let's see whether things get blocked or we can slowly move forward.
+Justin speaks of defining sets of resources and that solution fits very nicely.
+
+Justin: The way it's designed, clients should be able to read and write it on common data structures.
+In the current implementation that we've done, we look at data grants and then know how to apply the permissions on resources depending on the structures of resources and data. The authorization agent that understands data structures knows how to apply permissions.
+The authorization server who has access could look at the grant directly and be able to say whether you have access or not.
+Sort of leapfrogging having acp policies applying data grants.
+
+Elf: If you have an algorithm translating the data grants could even not be required.
+If using data grants, you need to be warry of keeping a single source of truth.
+
+Henry: You're bringing a whole bunch of architecture changes. I think it's a little bit like where are the grants... We'll have to look at it and see whether it's better than WAC.
+There's a whole thing about why we may want ACP and a lot of the assumptions for justifying it might have been wrong.
+We should move on to use cases and there, the precise explanation of how things are going, it's difficult to say just by listening to the argument.
+
+Elf: I'm a little worried that the use cases started as WAC UCRs. But Henry, you're right we should compare the solutions. I think it's important to keep things simple from a application development perspective.
+It's a big change of assumptions to have any application have to understand control rules, because I don't think anything that isn't dedicated shouldn't have to care about that.
```

Comment:
```suggestion
It's a big change of assumptions to have any application have to understand control rules, because I don't think anything that isn't dedicated should have to care about that.
```